### PR TITLE
Added metrics as sketched on https://gist.github.com/jdegoes/6570d5f8…

### PR DESCRIPTION
**Pull request by toxicafunk**
Add metrics according to Gist and moved pre-existing code to the `server` package

### CODE REVIEW

**Comment by jczuchnowski** [comment]
:+1:

File: src/main/scala/zio/zmx/server/ZMXClient.scala [hunk 1]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-package zio.zmx
+package zio.zmx.server
```
Comment (softinio) [inline src/main/scala/zio/zmx/server/ZMXClient.scala:17]: Out of interest why this change in package?

Comment (toxicafunk) [reply src/main/scala/zio/zmx/server/ZMXClient.scala:17]: Just to separate implementation between the protocol server (`server`) and the one for `metrics`. We can change that if you prefer some other way.

Comment (softinio) [reply src/main/scala/zio/zmx/server/ZMXClient.scala:17]: Good idea, got it now .... all good, happy with it. Ready for review?

Comment (softinio) [reply src/main/scala/zio/zmx/server/ZMXClient.scala:17]: rebase
